cunumeric/array.py

```diff
@@ -3811,6 +3854,16 @@ def _maybe_convert(self, dtype: np.dtype[Any], hints: Any) -> ndarray:
         copy._thunk.convert(self._thunk)
         return copy
 
+    def _warn_and_convert(self, dtype: np.dtype[Any]) -> ndarray:
```

Comment:
There's a couple more places in array.py and module.py where this pattern appears. Could you also convert those to use `_warn_and_convert`?

Reply:
sure

Reply:
On second thought, most of those places were not emitting warnings before, and I don't want to reason in this PR about whether each of those places should have warnings or not, so I just pushed a change to just use `astype` where the original code was not already throwing a warning.

Reply:
Thank you for doing that. I was expecting follow-up on this after I pushed the changes